compiler/il/OMRNode.cpp

```diff
@@ -3911,13 +3911,25 @@ OMR::Node::createStoresForVar(TR::SymbolReference * &nodeRef, TR::TreeTop *inser
                   }
                else
                   {
-                  newArrayRef = comp->getSymRefTab()->
-                     createTemporary(comp->getMethodSymbol(), TR::Address);
+                  TR::Node *arrayObjectNode = NULL;
+                  TR::Node *arrayLoadNode = NULL;
 
-                  TR::Node *newStore = TR::Node::createStore(newArrayRef, child);
+                  if (child->isDataAddrPointer())
+                     arrayObjectNode = child->getFirstChild();
+
+                  newArrayRef = comp->getSymRefTab()->createTemporary(comp->getMethodSymbol(), TR::Address);
+                  TR::Node *newStore = TR::Node::createStore(newArrayRef, arrayObjectNode ? arrayObjectNode : child);
```

Comment:
This tests whether `arrayObjectNode` is non-null in deciding whether to use `arrayObjectNode` or `child`, but the condition on the if statement below uses `child->isDataAddrPointer()` again.  It might be easier to read if both used the same condition - I have a slight preference for `child->isDataAddrPointer()` to make it more obvious `arrayObjectNode` will be operated set or referenced in all three places.